syncer/main.py

```diff
@@ -54,137 +127,369 @@ def __init__(self, webinterface, connector):
+                                                                    unit=remote_sensor.get('unit'),
+                                                                    name=f"{gw_name}/{remote_sensor.get('name')}")
+                            sensor_mapping[remote_sensor_id] = sensor
+                except Exception as ex:
+                    logger.exception(f'Could not load sensor mapping for GW with ip {ip}: {ex}')
+
+            self._gateways[ip]["sensor_mapping"] = sensor_mapping
+
+            # Global output_mapping variable because this is local 2 remote, thus if local output changes -> mapping to
+            # remote outputs will happen. One specific local output changes to different remote output changes
+            local_output_conf = json.loads(self.webinterface.get_output_configurations()).get('config')
+
+            for entry in gateway.get('outputs', []):
+                try:
+                    # Organise local output data and their mapping with one or more remote outputs by making use of a dict
+                    local_id = int(entry.get('local_output_id', -1))
+                    remote_id = int(entry.get('remote_output_id', -1))
+                    if local_id < 0 or remote_id < 0:
+                        raise RuntimeError("Please give valid output ids")
+                    if local_output_conf[local_id].get("type") == 127:
+                        raise RuntimeError(
+                            f"Skipped because {local_id} is a shutter, please include it in the shutter config instead")
+
+                    state = json.loads(self.webinterface.get_output_status()).get("status")[local_id]
+                    config = local_output_conf[local_id]
+                    if local_id not in output_mapping.keys():
+                        output_mapping[local_id] = {
+                            'remote_outputs': [],
+                            'config': config,
+                            'state': state
+                        }
+                    local_name = config.get('name') if config.get("name") != "" else local_id
+
+                    output_mapping[local_id]['remote_outputs'].append({
+                        'remote': remote_id,
+                        'gw': ip,
+                        'name': f"{self._name}/{local_name}"
+                    })
+
+                except Exception as ex:
+                    logger.exception(f'Could not load output mapping for GW with ip {ip}: {ex}')
+                    continue
+
+                try:
+                    self.update_remote_output_config_and_state(ip, remote_id, state, remote_output_conf=remote_output_conf, local_output_name=local_name)
+                except Exception as ex:
+                    logger.info(f"Error while reading plugin output configuration: {ex}")
+                    continue
+
+            # Global shutter mapping
+            local_shutter_confs = json.loads(self.webinterface.get_shutter_configurations()).get('config')
```

Comment:
no try/except block here? :)

Reply:
This should always work, otherwise the GW is not running as should be.

Reply:
Same has been done with connectors, they should also work, otherwise something in the GW is faulty